src/components/Settings/Settings.js

```diff
@@ -62,7 +32,19 @@ const Settings = () => {
       </div>
       <div className="settings__control">
         <h3>All addresses:</h3>
-        {addresses}
+        {isLoading && !error && <LoadingSpinner />}
+        {!isLoading && error && (
+          <Error>{error.message || 'Sth went wrong'}</Error>
+        )}
+        {!isLoading &&
+          !error &&
```

Comment:
I think you don't have check both `isLoading` and `error` in every of those cases.

For example to show `<LoadingSpinner />` it is not necessary to check `error`
And the same situation is with displaying error: you can check only `error`

BTW simpler and cleaner solution would something like:
```
if(error) {
 return <Error />
}

if(isLoading){
  return <LoadingSpinner />
}

return (
  data.map((address) => (
    <AddressItem
       id={address.id}
       address={address.address}
       key={address.id}
     />
  ))}
)
```